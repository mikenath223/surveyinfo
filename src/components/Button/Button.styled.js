import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const BaseButton = styled(Button)`
  a {
    text-decoration: none;
    color: #fff;
  }
  .source {
    color: green;
  }
`