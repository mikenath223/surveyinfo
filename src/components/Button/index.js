import styled from 'styled-components';
import theme from 'styled-theming';
import buttonBackgound from 'styles/global';

const backgroundColor = theme('mode', buttonBackgound);

export const Button = styled.button`
  background-color: ${backgroundColor};
`;