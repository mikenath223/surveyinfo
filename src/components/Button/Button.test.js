import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import Button from './index';

const onClickMock = jest.fn();

describe('Button', () => {
  let wrapper, getText;
  beforeEach(() => {
    const { container, getByText } = render(
      <Button onClick={onClickMock}>Testing</Button>
    );

    wrapper = container;
    getText = getByText;
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to click button component', () => {
    fireEvent.click(getText(/Testing/));

    expect(onClickMock).toBeCalled();
  });
});
