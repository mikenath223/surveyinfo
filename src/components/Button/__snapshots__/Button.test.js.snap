// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
.c0.source {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: green;
}

.c0.link {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <button
    class="c0 btn btn-primary"
    type="button"
  >
    Testing
  </button>
</div>
`;
